chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/jwt/JSONWebKeyRsaJoseJImpl.java

```diff
@@ -26,95 +26,91 @@
 
-	@Autowired Environment env;
-	RsaJsonWebKey rsaJsonWebKey = null;
-	
-	Logger logger = LogManager.getLogger(JSONWebKeyRsaJoseJImpl.class.getName());
-	
-	public JSONWebKeyRsaJoseJImpl() {
-	}
-	
-	@PostConstruct
-	public void createOrLoadKey(){
-		String keyLocation = env.getProperty("keyLocation");
-		
-		if (rsaJsonWebKey == null){
-			
-			try {
-				loadSavedKey(keyLocation);
-			} catch (ClassNotFoundException | IOException e1) {
-				
-				logger.error("Key not found or error loading key. Creating new key. " , e1);
-				
-				try {
-			        RsaKeyUtil keyUtil = new RsaKeyUtil();
-			        KeyPair keyPair;
-					keyPair = keyUtil.generateKeyPair(2048);
-			        rsaJsonWebKey = (RsaJsonWebKey) PublicJsonWebKey.Factory.newPublicJwk(keyPair.getPublic());
-			        rsaJsonWebKey.setPrivateKey(keyPair.getPrivate());
-			        saveKey(keyLocation);
-				} catch (JoseException e) {
-					
-					logger.error("Error creating key: " , e1);
-					
-					throw new RuntimeException(e);
-				}
-			}
-		}
-	}
-	
-	@Override
-	public String getKeyId() {
-		return rsaJsonWebKey.getKeyId();
-	}
-
-	@Override
-	public String getAlgorithm() {
-		return rsaJsonWebKey.getAlgorithm();
-	}
-
-	@Override
-	public Key getKey() {
-		return rsaJsonWebKey.getKey();
-	}
-
-	@Override
-	public PrivateKey getPrivateKey() {
-		return rsaJsonWebKey.getPrivateKey();
-	}
-
-	@Override
-	public PublicKey getPublicKey() {
-		return rsaJsonWebKey.getPublicKey();
-	}
-
-	
-	public void saveKey(String keyPairPath){
-		
-		try {
-			
-			File file = new File(keyPairPath);
-			file.getParentFile().mkdirs();
-			FileOutputStream fileOut = new FileOutputStream(file);
-			ObjectOutputStream out = new ObjectOutputStream(fileOut);
-			out.writeObject(rsaJsonWebKey);
-			out.close();
-			fileOut.close();
-			
-		} catch (IOException e) {
-			logger.error("Error saving key: " , e);
-		}
-	}
-	
-	public void loadSavedKey(String keyPairPath) throws IOException, ClassNotFoundException {
-		
-		FileInputStream fileIn = new FileInputStream(keyPairPath);
-		ObjectInputStream is = new ObjectInputStream(fileIn);
-		
-		rsaJsonWebKey = (RsaJsonWebKey) is.readObject();
-		
-		is.close();
-		fileIn.close();
-		
-	}
+    @Autowired
+    Environment env;
+    RsaJsonWebKey rsaJsonWebKey = null;
+
+    Logger logger = LogManager.getLogger(JSONWebKeyRsaJoseJImpl.class.getName());
+
+    public JSONWebKeyRsaJoseJImpl() {
+    }
+
+    @PostConstruct
+    public void createOrLoadKey() {
+        String keyLocation = env.getProperty("keyLocation");
+
+        if (rsaJsonWebKey == null) {
+
+            try {
+                loadSavedKey(keyLocation);
+            } catch (ClassNotFoundException | IOException e1) {
+
+                logger.error("Key not found or error loading key. Creating new key. ", e1);
+
+                try {
+                    RsaKeyUtil keyUtil = new RsaKeyUtil();
+                    KeyPair keyPair;
+                    keyPair = keyUtil.generateKeyPair(2048);
```

Comment:
2048 looks like a magic number

Reply:
It looks like 2048 bits for a key length is pretty standard. Unless there's a good reason we should change it to 4096 I will keep that the same.

Reply:
I think "magic number" just means it's a number in the midst of code rather than a constant field